integration_tests/e2e/manualUnusedDeduction.cy.ts

```diff
@@ -0,0 +1,75 @@
+import HubPage from '../pages/hub'
+import FormPage from '../pages/form'
+import AddManualUnusedDeductionsPage from '../pages/unused-deductions/addManualUnusedDeduction'
+import ReviewManualUnusedDeductionsPage from '../pages/unused-deductions/reviewManualUnusedDeduction'
+import EditManualUnusedDeductionsPage from '../pages/unused-deductions/editManualUnusedDeduction'
+
+context('Enter Unused Deductions', () => {
+  beforeEach(() => {
+    cy.task('reset')
+    cy.task('stubSignIn')
+    cy.task('stubManageUser')
+    cy.task('stubGetPrisonerDetails')
+    cy.task('stubGetUserCaseloads')
+    cy.task('stubGetAdjustments')
+    cy.task('stubGetTaggedBailAdjustment')
+    cy.task('subAdaDetailsNoIntercept')
+    cy.task('stubGetSentencesAndOffences')
+    cy.task('stubCreateAdjustment')
+    cy.task('stubUpdateAdjustment')
+    cy.task('stubDeleteTaggedBailAdjustment')
+    cy.task('stubComponents')
+    cy.task('stubGetUnusedDeductionsCalculationResultUnsupported')
+    cy.task('stubSetUnusedDaysManually')
+  })
+
+  it('Add Manual Unused Deduction', () => {
+    cy.signIn()
+    const hub = HubPage.goTo('A1234AB')
+    hub.manualUnusedDeductionsLink().click()
+    AddManualUnusedDeductionsPage.verifyOnPage(AddManualUnusedDeductionsPage)
+    const form = FormPage.verifyOnPage<FormPage>(FormPage, 'Enter the number of unused deductions')
+    form.enterDays('10')
+    form.submitButton().click()
+    const reviewManualUnusedDeductionsPage = ReviewManualUnusedDeductionsPage.verifyOnPage(
+      ReviewManualUnusedDeductionsPage,
+    )
+    reviewManualUnusedDeductionsPage.submit().click()
+    hub.successMessage().contains('10 days of unused deductions have been saved')
+  })
+})
+
+context('Edit Unused Deductions', () => {
+  beforeEach(() => {
```

Comment:
Can these tests be combined so that you only have one `beforeEach()`?

Also I think you should review all the stub's here. I don't think you need all of them.

Reply:
The edit and delete tests need to use a different stub from the add. Is there a way to use different stubs within 1 `beforeEach()`?

Reply:
So don't include the stubs which are specific to the test in the before each. Put the the stubs specific to each test at the start of the test.